src/renderer/entities/network/lib/networkService.ts

```diff
@@ -1,388 +1,115 @@
-      subscribeDisconnected(chainId, provider.instance);
-      subscribeError(chainId, provider.instance, onAutoBalanceError);
-    } else {
-      updateConnectionState(chainId, {
-        activeNode: node,
-        connectionType: type,
-        connectionStatus: ConnectionStatus.ERROR,
-      });
-    }
-  };
-
-  const validateRpcNode = (chainId: ChainId, rpcUrl: string): Promise<RpcValidation> => {
-    return new Promise((resolve) => {
-      const CachedWsProvider = createCachedProvider(WsProvider, chainId, getMetadata);
-
-      const provider = new CachedWsProvider(rpcUrl);
-
-      provider.on('connected', async () => {
-        let isNetworkMatch = false;
-        try {
-          const api = await ApiPromise.create({ provider, throwOnConnect: true, throwOnUnknown: true });
-          isNetworkMatch = chainId === api.genesisHash.toHex();
-
-          api.disconnect().catch(console.warn);
-        } catch (error) {
-          console.warn(error);
-        }
-        provider.disconnect().catch(console.warn);
-        resolve(isNetworkMatch ? RpcValidation.VALID : RpcValidation.WRONG_NETWORK);
-      });
-
-      provider.on('error', async () => {
-        try {
-          await provider.disconnect();
-        } catch (error) {
-          console.warn(error);
-        }
-        resolve(RpcValidation.INVALID);
-      });
-    });
-  };
-
-  const addRpcNode = async (chainId: ChainId, rpcNode: RpcNode): Promise<void> => {
-    const connection = connections[chainId];
-    if (!connection) return;
-
-    await updateConnectionState(chainId, {
-      customNodes: (connection.connection.customNodes || []).concat(rpcNode),
-    });
-  };
-
-  const updateRpcNode = async (chainId: ChainId, oldNode: RpcNode, newRpc: RpcNode): Promise<void> => {
-    const connection = connections[chainId];
-    if (!connection) return;
-
-    await updateConnectionState(chainId, {
-      customNodes: (connection.connection.customNodes || []).map((node) => {
-        if (node.url === oldNode.url && node.name === oldNode.name) return newRpc;
-
-        return node;
-      }),
-    });
-  };
-
-  const removeRpcNode = async (chainId: ChainId, rpcNode: RpcNode): Promise<void> => {
-    const connection = connections[chainId];
-    if (!connection) return;
+    return new CachedScProvider(Sc, knownChainId);
+  } else {
+    throw new Error('Parachains do not support Substrate Connect yet');
+  }
+};
 
-    await updateConnectionState(chainId, {
-      customNodes: (connection.connection.customNodes || []).filter((node) => node.url !== rpcNode.url),
-    });
-  };
+const createWebsocketProvider = (
+  rpcUrls: string[],
+  chainId: ChainId,
+  getMetadata: (chainId: ChainId) => Promise<Metadata | undefined>,
+): ProviderInterface => {
+  // TODO: handle limited retries provider = new WsProvider(node.address, 5000, {1}, 11000);
+  const CachedWsProvider = createCachedProvider(WsProvider, chainId, getMetadata);
```

Comment:
I saw that we're creating wrapping both sc and ws providers in cached provider separately. Might be worth to wrap in a single place instead 

``` typescript
const provider = createWebsocketProvider/createScProvider()
const cachedMetadataProvider = createCachedProvider(provider,...)
```